lib/Quantization/Quantization.cpp

```diff
@@ -27,6 +28,112 @@ using llvm::cast;
 namespace glow {
 namespace quantization {
 
+/// This class produces a quantized function based on a provided profile.
+class FunctionQuantizer : public FunctionConverter {
+protected:
+  /// \see FunctionConverter::getTargetTypeForOutput.
+  /// Get the quantized type for \p out if any.
+  TypeRef getTargetTypeForOutput(const NodeValue &out) const override {
+    if (out.getElementType() != ElemKind::FloatTy) {
+      return nullptr;
+    }
+
+    const std::string nodeOutputName =
+        NodeQuantizationInfo::generateNodeOutputName(out.getNode()->getName(),
+                                                     out.getResNo());
+    auto outTQPIt = nodeToTQP_.find(nodeOutputName);
+    assert(outTQPIt != nodeToTQP_.end() &&
+           "Missing quantization params for a node");
+
+    const TensorQuantizationParams &TQP = outTQPIt->second;
+    return mod_.uniqueType(ElemKind::Int8QTy, out.dims(), TQP.scale,
+                           TQP.offset);
+  }
+
+  /// \see FunctionConverter::getTargetTypeForInput.
+  /// Get the quantized type for the \p idx-th argument of \p use, if any.
+  TypeRef getTargetTypeForInput(const Node &use, unsigned idx) const override {
+    NodeValue val = use.getNthInput(idx);
+
+    // Do not quantize non floating point type, e.g., Index type.
+    if (val.getElementType() != ElemKind::FloatTy) {
+      return nullptr;
+    }
+
+    std::string nodeOutputName = NodeQuantizationInfo::generateNodeOutputName(
+        val.getNode()->getName(), val.getResNo());
+    auto valTQPIt = nodeToTQP_.find(nodeOutputName);
+    assert(valTQPIt != nodeToTQP_.end() &&
+           "Missing quantization params for a node");
+
+    const TensorQuantizationParams &TQP = valTQPIt->second;
+    return mod_.uniqueType(ElemKind::Int8QTy, val.dims(), TQP.scale,
+                           TQP.offset);
+  }
+
+  /// \see FunctionConverter::canConvert.
+  /// Only convert nodes that use floating point types and that
+  /// weren't specifically marked as to-ignore with doNotQuantizeKinds_.
+  bool canConvert(const Node &node) const override;
```

Comment:
Any particular reason this (and morphNode and postProcessing) are defined out-of-line?  It would be nice to be consistent with that.

Reply:
Not really, it was more a judgement call like "hmm, those are going to be big". Turns out they are not really.

Reply:
Done